test/useObservable.test.tsx

```diff
@@ -120,4 +120,25 @@ describe("is used to keep observable within component body", () => {
         fireEvent.click(div)
         expect(div.textContent).toBe("initial - 10later - 20")
     })
+
+    it("should work with lazy initialization", () => {
+        const TestComponent = () => {
+            const obs = useObservable(() => ({
+                x: 1,
+                y: 2

```

Comment:
Please add a getter and action here just to be safe on the type side. I tried what inferrence problem there is without overloading and it mostly works except that getter in the above test.

Reply:
added tests for action and getter in new commit.